pallets/subtensor/src/staking.rs

```diff
@@ -158,12 +158,25 @@ impl<T: Config> Pallet<T> {
         );
 
         // --- 6. Ensure we don't exceed stake rate limit
-        let stakes_this_interval = Self::get_stakes_this_interval_for_coldkey_hotkey(&coldkey, &hotkey);
+        let stakes_this_interval =
+            Self::get_stakes_this_interval_for_coldkey_hotkey(&coldkey, &hotkey);
         ensure!(
             stakes_this_interval < Self::get_target_stakes_per_interval(),
             Error::<T>::StakeRateLimitExceeded
         );
 
+        // --- 7. If this is a nomination stake, check if total stake after adding will be above
+        // the minimum required stake.
+        if !Self::coldkey_owns_hotkey(&coldkey, &hotkey) {
+            let total_stake_after_add =
+                Stake::<T>::get(&hotkey, &coldkey).saturating_add(stake_to_be_added);
+
+            ensure!(
+                total_stake_after_add >= NominatorMinRequiredStake::<T>::get(),
+                Error::<T>::NomStakeBelowMinimumThreshold
```

Comment:
can we perhaps add some more documentation as to why we're ensuring that coldkey does NOT own the hotkey?